src/main/java/io/vertx/core/impl/VertxImpl.java

```diff
@@ -869,19 +875,14 @@ private class InternalTimerHandler implements Handler<Void>, Closeable, Runnable
     private final boolean periodic;
     private final long timerID;
     private final ContextInternal context;
-    private final java.util.concurrent.Future<?> future;
+    private java.util.concurrent.Future<?> future;
```

Comment:
This need to be volatile, probably. 
You can use a `StoreStore` barrier (ie lazySet) or a full volatile here (safer).

Reply:
I don't think it is necessary, the thread storing the field will then pass the timerID to another thread that will need to lookup the object in the timeout map.